link.go

```diff
@@ -527,6 +537,14 @@ func (l *link) muxReceive(fr performTransfer) error {
 	l.deliveryCount++
```

Comment:
is is still correct to update the deliveryCount and linkCredit states if what we got in the select is a close?